docs/smart_contracts.rst

```diff
@@ -0,0 +1,142 @@
+Raiden Network Smart Contracts
+==============================
+
+Smart Contract Architecture
```

Comment:
The architecture could be part of the spec itself so perhaps not include them (at least for now) in their own section here?

Reply:
Ok, if you think it fit's better over there I'll change the heading.